### DISCUSSION TITLE
Difference between `interface_str` and `behavior_modules_str`

### DISCUSSION THREAD

**Discussion by hash-ir** [AdaptiveMotorControlLab/AmadeusGPT#9] [Q&A] [answered]
Hi,

I am a bit confused about what the API docs really mean. In this regard, I have the following questions:

1. In the system prompt, there are two types of documentation provided to GPT 3.5, `interface_str` and `behavior_modules_str`. What is the difference between the two? 
2. Also, in the Appendix under **Examples of API core behavioral modules**, an example class is provided. Is it one of the behavior modules or part of the `interface_str`?
3. What is the difference between the `Methods` specified in the class docstring and the actual methods of the class that have the example functions? e.g. difference between `get_object_names()` and `animals_state_events`?

Thank you in advance for considering my questions. Looking forward to seeing more about AmadeusGPT!

**Comment by yeshaokai** [comment, marked as answer]
@hash-ir  Thanks so much for interest in our work. I am sorry I get back to you so late (didn't get notification on time)

1) and 2) interface_str corresponds to "core_modules " from our paper and "behavior_modules_str" corresponds to integration modules from our paper.

3) there is no difference. When testing Amadeus with GPT-3.5, we had very limited context window so I wrote detailed docstr for functions that have complicated usage and very minimal docstr for easy, intuitive functions to save space for the short-term memory

Now the code is released and feel free to explore and ask further questions if there are
